.github/scripts/setup_runners.sh

```diff
@@ -0,0 +1,120 @@
+                          --key-name $SETUP_KEYNAME --security-groups $SETUP_SECURITY_GROUP \
+                          --iam-instance-profile Name=$SETUP_IAM_NAME \
+                          --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=$instance_name1}]" > /dev/null 2>&1; echo $?
+    sleep 1
+  done
+
+  echo ""
+  echo "Sleep for 120 seconds for EC2 instances to start running"
+  echo ""
+
+  sleep 120
+
+  # Setup VMs to register as runners
+  for instance_name2 in $SETUP_INSTANCE
+  do
+    echo "[${instance_name2}]: Make change of the runner hostname"
+    aws ssm send-command --targets Key=tag:Name,Values=$instance_name2 --document-name "AWS-RunShellScript" \
+                         --parameters '{"commands": ["#!/bin/bash", "sudo hostnamectl set-hostname '${instance_name2}'"]}' \
+                         --output text > /dev/null 2>&1; echo $?
+
+    echo "[${instance_name2}]: Get runner token and bootstrap on Git"
+    instance_runner_token=`curl --silent -H "Authorization: token ${SETUP_TOKEN}" --request POST "${GIT_URL_API}/${GIT_URL_REPO}/actions/runners/registration-token" | jq -r .token`
+    aws ssm send-command --targets Key=tag:Name,Values=$instance_name2 --document-name "AWS-RunShellScript" \
+                         --parameters '{"commands": ["#!/bin/bash", "sudo su - '${SETUP_AMI_USER}' -c \"cd actions-runner && ./config.sh --unattended --url '${GIT_URL_BASE}/${GIT_URL_REPO}' --labels '${instance_name2}' --token '${instance_runner_token}' && nohup ./run.sh &\""]}' \
+                         --output text > /dev/null 2>&1; echo $?
+    sleep 5
+  done
+
+  echo ""
+  echo "Wait for 60 seconds for runners to bootstrap on Git"
+  echo ""
+
+  sleep 60
+
+  echo ""
+  echo "All runners are online on Git"
+  echo ""
+fi
+
+
+###################################################
+# Terminate / Delete instances and remove runners #
+###################################################
+if [ "$SETUP_ACTION" = "terminate" ]
+then
+
+  echo ""
+  echo "Terminate / Delete instances and remove runners [${SETUP_INSTANCE}]"
+  echo ""
+
+  for instance_name3 in $SETUP_INSTANCE
+  do
+    instance_runner_id_git=`curl --silent -H "Authorization: token ${SETUP_TOKEN}" --request GET "${GIT_URL_API}/${GIT_URL_REPO}/actions/runners" | jq ".runners[] | select(.name == \"${instance_name3}\") | .id"`
+    echo "[${instance_name3}]: Unbootstrap runner from Git"
+    curl --silent -H "Authorization: token ${SETUP_TOKEN}" --request DELETE "${GIT_URL_API}/${GIT_URL_REPO}/actions/runners/${instance_runner_id_git}"; echo $?
+
+    instance_runner_id_ec2=`aws ec2 describe-instances --filters "Name=tag:Name,Values=$instance_name3" | jq -r '.Reservations[].Instances[] | select(.State.Code == 16) | .InstanceId'` # Only running instances
+    echo "[${instance_name3}]: Remove tags Name"
+    aws ec2 delete-tags --resources $instance_runner_id_ec2 --tags Key=Name > /dev/null 2>&1; echo $?
```

Comment:
why do we need delete the tags before terminating the instance

Reply:
You have to, if not, there will exist multiple instances in termination state with the same tags as newly created instances. Even though I have setups to detect instance with tags that are running, it is still safer to remove the tags if you dont need the instance anymore.